crates/circuit/src/circuit_data.rs

```diff
@@ -430,6 +335,32 @@ impl CircuitData {
         self.clbits.len()
     }
 
+    /// Return the number of unbound compile-time symbolic parameters tracked by the circuit.
+    pub fn num_params(&self) -> usize {
```

Comment:
This is the number of `Parameter` objects, right? Should we name this consistently with the current `QuantumCircuit.num_parameters`?
```suggestion
    pub fn num_parameters(&self) -> usize {
```

Reply:
Yeah, yours is better and in keeping with the rest of the terminology in Rust space.  I'll update it everywhere.

Reply:
Done in 8c554ba7e0.